# Homework 5
###### by Mher Movsisyan
---

## Problem 1  
  
| Year | Company 1 | Company 2 |  
| ---- | --------- | --------- |  
| 1 | 102 | 173 |  
| 2 | 116 | 121 |  
| 3 | 132 | 156 |  
| 4 | 145 | 133 |  
| 5 | 126 | 179 |  
| 6 | 126 | 129 |  
| 7 | 129 | 153 |  
| 8 | 111 | 157 |  
| 9 | 124 | 150 |  
| 10 | 156 | 124 |  

You have observed the profit (mln $) of two companies over 10
consecutive years. You want to test whether there is sufficient
evidence to indicate that the profit of the `Company 1` exceed 
to profit of the `Company 2`. Conduct the Sign test for 
Matched-Pairs Experiment to answer the following questions.

a) Define the Null and Alternative hypotheses (2 points)

Answer:  
H<sub>0</sub>: `Company 1`'s profit distribution is less than or equal to `Company 2`'s profit distribution in terms of <b>location</b>.  
H<sub>Alt</sub>: H<sub>0</sub> is false.

b) What is the rejection region for 1% significance level? (10 points)

Answer:

In [43]:
from scipy.stats import binom, chi2, spearmanr

[1 - binom.cdf(i, 10, 0.5) for i in range(9, 7, -1)]

[0.0009765625, 0.0107421875]

10 or 9 observations of greater profit relative to `Company 2`

c) Calculate the p-value for the test? (5 points)

In [18]:
import numpy as np

n_pos_obs = (np.array([
    102 - 173,
    116 - 121,
    132 - 156,
    145 - 133,
    126 - 179,
    126 - 129,
    129 - 153,
    111 - 157,
    124 - 150,
    156 - 124
]) > 0).sum()

binom.cdf(n_pos_obs, 10, 0.5)

0.0546875

d) Why it might be inappropriate to use the t-test to analyze the data? (5 points)

Answer:  
Because we have too few samples, we don't know the shape of the profit distribution (t-test assumes normal)

e) Do you reject the Null Hypothesis? Why? (3 points)

Answer:  
No because `p = 0.054` <b>></b> `0.01 = alpha`

### Problem 2 (20 points)  
Given in Table 2 are weights of horses for samples of two species. Six horses of Auxois and seven of Campeiro.  

| Auxois | Campeiro |  
| ------ | -------- |  
| 390 | 400 |   
| 460 | 550 |   
| 400 | 640 |   
| 420 | 700 |   
| 418 | 620 |   
| 435 | 600 |   
| | 53 |  

a) Do the data present sufficient evidence to indicate that the 
distributions of weights differ for the two species? Use the test 
based on the Mann-Whitney U statistic with 𝛼 =0.05 (10 points)

b) Give the approximate p-value associated with the test. (10 points)

Answer:  
Ordered data = `53, 390, 400, 400, 418, 420, 435, 460, 550, 600, 620, 640, 700`  
Assigned ranks:  
  
| 53 | 390 | 400 | 400 | 418 | 420 | 435 | 460 | 550 | 600 | 620 | 640 | 700 |  
| -- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- |  
| 1  |   2 | 3.5 | 3.5 |   5 |   6 |   7 |   8 |   9 |  10 |  11 |  12 | 13  |  
| C  |   A |  C  | A   | A   | A   | A   | A   |  C  |  C  |  C  |  C  |  C  |  


$$ U_A = 6 * 7 + \frac{6 * 7}{2} - (2 + 3.5 + 5 + 6 + 7 + 8) = 31.5 $$
$$ U_C = 6 * 7 + \frac{7 * 8}{2} - (1 + 3.5 + 9 + 10 + 11 + 12 + 13) = 10.5 $$
$$ U = min(31.5, 10.5) = 10.5 $$
$$ U_{expected} = \frac{6 * 7}{2} = 21 $$

$$ \sigma_{U} = \sqrt{\frac{6 * 7}{13 * 12}} * \sqrt{\frac{13^3 - 13}{12} - 0} = 7 $$

$$ z = \frac{U - U_{expected}}{\sigma_U} = -1.5 $$  
$$ p = 0.06681 $$

We do not reject the null hypothesis of the two species having the same weight distribution location at `alpha = 0.05`.

### Problem 3 (25 points)  
a) Using the data in the Table 3, conduct one-tailed Wilcoxon 
Signed-Rank Test for Matched-Pairs experiment. Calculate 
differences as B - A. Interpret the results (10 points)

In [19]:
data = [[54, 53],
[25, 88] ,
[63, 83] ,
[130, 61] ,
[174, 47] ,
[61, 148] ,
[112, 164] ,
[53, 149] ,
[168, 98] ,
[81, 111]]

diffs = np.array([(i[1] - i[0]) for i in data])
diffs

array([  -1,   63,   20,  -69, -127,   87,   52,   96,  -70,   30])

In [20]:
indices = np.argsort(np.absolute(diffs))
diffs[indices]

array([  -1,   20,   30,   52,   63,  -69,  -70,   87,   96, -127])

1 + 6 + 7 + 10 = 24  
2 + 3 + 4 + 5 + 8 + 9 = 31  
31 - 24 = 7  
  
Having a t-stat of 7 tells us that we have less than 0.025% chance of having a type 1 error.

b) Conduct two-tailed Wilcoxon Signed-Rank Test to detect 
differences in distributions differences. Interpret the results. (15 
points)

Answer:  
Same as the previous (`3.a`) problem except now we have less than 0.05% chance of having a type 1 error.

### Problem 4 (20 points)  
There are three different student groups of student. The table 3 contains GPA of students in 
each Group. Assuming that the samples are independent.
  
| Student Group_ID | Group 1 | Group 2 | Group 3 |  
| ---------------- | ------- | ------- | ------- |  
| 1                | 3.5     | 2.5     | 2.6     |  
| 2                | 2.5     | 1.9     | 1.7     |  
| 3                | 1.7     | 2.6     | 2.7     |  
| 4                | 3.1     | 3.4     | 2.2     |  
| 5                | 3.9     | 3.0     | 2.8     |  
| 6                | 3.5     |         |         |  
  
a) Do the data in the table 3 provide evidence 
that the probability distributions of the GPA 
are differ between groups for at least two of 
the groups? Use 𝛼 = 0.05 (10 points)  
b) Calculate p-value. (10 points)

Answer:  
We do a Krusal-Wallis test

In [21]:
g1 = [3.5,2.5,1.7,3.1,3.9,3.5]
g2 = [2.5,1.9,2.6,3.4,3.0]
g3 = [2.6,1.7,2.7,2.2,2.8]

ranks = [1.5, 1.5, 3.0, 4.0, 5.5, 5.5, 7.5, 7.5, 9.0, 10.0, 11.0, 12.0, 13.0, 14.5, 14.5, 16.0]
list(zip(ranks, sorted(g1 + g2 + g3)))

[(1.5, 1.7),
 (1.5, 1.7),
 (3.0, 1.9),
 (4.0, 2.2),
 (5.5, 2.5),
 (5.5, 2.5),
 (7.5, 2.6),
 (7.5, 2.6),
 (9.0, 2.7),
 (10.0, 2.8),
 (11.0, 3.0),
 (12.0, 3.1),
 (13.0, 3.4),
 (14.5, 3.5),
 (14.5, 3.5),
 (16.0, 3.9)]

In [37]:
rs1 = 1.5 + 5.5 + 14.5 + 14.5 + 16 + 12
rs2 = 5.5 + 3 + 7.5 + 13 + 11
rs3 = 7.5 + 1.5 + 9 + 4 + 10

rsa1 = rs1 / 6
rsa2 = rs2 / 5
rsa3 = rs3 / 5

V = sum([[5, 5, 6][i] * np.power([rsa1, rsa2, rsa3][i] - np.sum(ranks)/len(ranks), 2) for i in range(2)])
H = (12 * V) / (16 * 17)
H

1.0906862745098034

In [42]:
1 - chi2.cdf(H, df=2)

0.5796428521543362

We fail to reject the null hypothesis, our p value is 0.5796

### Problem 5 (10 points)
a) Calculate the Spearman’s rank correlation coefficient
between house price and the area. (5 points)  
b) Use the test based on the rank correlation
coefficient to determine whether there is a
significant positive relationship between price and
the area. Use 𝛼 = 0.1 (5 points)  

In [44]:
price = [89950,138950,87000,165000,210000,108000,89000,79000,124500,135000,105500,133650,83500,101000,151500,88500,198000,135000,79500,135050]
sqft = [917,1684,1800,1900,2000,1050,1057,954,1350,2134,1313,1671,1200,1314,1877,1132,2198,1525,1208,1450]

spearmanr(price, sqft)

SpearmanrResult(correlation=0.7679579332107139, pvalue=7.695060782041737e-05)

The coefficient is 0.768 and is significant at a = 0.1